domain/network/state/linklayer_merge.go

```diff
@@ -39,10 +41,17 @@ type mergeLinkLayerDevice struct {
 // It contains only the fields that are used to identify and merge the addresses
 type mergeAddress struct {
```

Comment:
... if you add the node UUID to this type.

That ends up making it almost identical to `getIpAddress`.